### DISCUSSION TITLE
Connecting to Memorystore Redis for Message Queue Implementation

### DISCUSSION THREAD

**Discussion by VinuthnaGangula** [taskforcesh/bullmq#2287] [Q&A] [answered]
Hello there,

I tried using BullMQ's Latest Version (4.14.0) to implement a Message Queue system for a chat-based application. For the Redis connection, I was trying to pass the connection data of Memorystore for Redis. I get the following error and I am assuming it is because, 'IORedis' is trying to use the 'Client' command, which is blocked by the Memorystore Redis instance.

```
ReplyError: ERR unknown command `client`, with args beginning with: `LIST`, 
    at parseError (/workspace/node_modules/redis-parser/lib/parser.js:179:12)
    at parseType (/workspace/node_modules/redis-parser/lib/parser.js:302:14) {
  command: { name: 'client', args: [ 'LIST' ] }
}
```

Is there a way to make this work or do I have to use a separate Redis server for implementing Queues?

Thanks in advance!

**Comment by manast** [comment, marked as answer]
You could try to create an instance of IORedis, then monkey patch the "client" method on it to do nothing and then pass that instance as a connection to BullMQ. (getWorkers API will not work after this hack).

**Reply by VinuthnaGangula** [reply]
Thanks for the reply. 

Creating an IORedis Client instance and passing it to Queue was exactly what I tried to do. And yes, I got this error while trying to use the very **getWorkers** method (For my case, everything else depends on this, ofcourse).

So in conclusion you mean, I need to use a Redis server for BullMQ that supports the above command?

**Reply by manast** [reply]
I am afraid that would be the case if you really need the getWorkers function.

**Reply by VinuthnaGangula** [reply]
Sure, thanks for clarifying.

Thanks for the support, really appreciate the quick response!
